---
title: "Detection of eco-evolutionary dynamics in metacommunities using Joint Species
  Distribution Models"
author:
  - Jelena H. Pantel^[Laboratoire Chrono-environnement,UMR 6249 CNRS-UFC, 16 Route de Gray, 25030 Besançon cedex, France, jelena.pantel@univ-fcomte.fr]
  - Ruben J. Hermann^[University of Duisburg-Essen, Universitätsstraße 5, 45141 Essen, Germany, ruben.hermann@uni-due.de]
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  bookdown::pdf_document2:
    toc: no
    fig_caption: yes
    latex_engine: xelatex
    extra_dependencies: ["float"]
    number_sections: true
  bookdown::html_document2: default
  word_document:
    reference_docx: docx_template.docx
header-includes:
- \usepackage{fancyhdr}
- \pagestyle{fancy}
- \fancyhead[L]{Author Pantel and Hermann}
- \fancyhead[R]{Detection of ecoevo dynamics}
- \usepackage{lineno}
- \linenumbers
linestretch: 2
link-citations: yes
linkcolor: blue
csl: the-american-naturalist.csl
bibliography: refs.bib
---

```{r setup,echo=FALSE,warning=FALSE,message=FALSE}
library(Hmsc)
```


\newpage

Title
===============================================================================

Detection of eco-evolutionary dynamics in metacommunities using Joint Species Distribution Models
    
Abstract
===============================================================================
  
XXXXX

Introduction
================================================================================

Analysis of biodiversity at large spatial scales can be complex, because numerous environmental properties, spatial structure and connectivity, interactions between species, as well as stochastic processes such as drift and priority effects operate and determine observed patterns. Statistical models in ecology have made tremendous progress in reflecting these numerous ecological processes at multiple scales. There are now a diversity of joint species distribution models [JSDMs; @Pollock2014] [@Pichler2021] or multi-species occupancy models [@Devarajan2020] [@Poggiato2021], which can incorporate these diverse components that structure variance in biological data. Many of these models increasingly have implementations in R packages (e.g. spOccupancy, [@Doser2022]), and have been used to measure the relative importance of different ecological drivers in fungi, birds, mammals, phytoplankton, fish, and other taxa, from local to continental scales (e.g. [@Abrego2020] [@Antao2022] [@Keppeler2022]).

There are some remaining drivers of community composition that aren't currently reflected in these models. The dynamics of populations and communities, and resulting biodiversity patterns, are also influenced by the genotypic and phenotypic properties of organisms as well. Although traits can be considered in some JSDMs, they typically require fixed values for species' traits ([@Ovaskainen2017HMSC] [@Tikhonov2020]; but see @Abrego2024). However, ecological dynamic processes can impact selection on genes and phenotypes, potentially resulting in interactions or feedback loops between ecological and evolutionary processes [@Lion2018] [@Barbour2022]. Although an increasing number of theoretical and empirical studies have evaluated the impact of evolutionary dynamics for community assembly and metacommunity dynamics [@Loeuille2008] [@Osmond2013] [@Pantel2015] [@Toju2017], analytical methods to consider this additional complexity lag behind the scope of data collected in eco-evolutionary studies. Some methods exist that ultimately apply an ANOVA framework to partition variance in traits or growth rates across contributions of ecological and evolutionary components ([@Hairston2005] [@Ellner2011] [@Govaert2016]). These methods tend to be cited more frequently for their potential than for their actual application to empirical data (but see [@terHorst2014] [@Hiltunen2014]), because this approach is ultimately limited by the same requirements that can limit application of ANOVA - data should follow a normal distribution, the processes that structure the target response variable must be linear and additive, and complicated structuring mechanisms must be reduced into one or a few target categories labeled as ‘ecology’ or ‘evolution’. Flexible models, which can vary depending on the mechanisms and structure of the diverse kinds of data that most studies of eco-evolutionary dynamics will collect, are needed for rigorous inference about eco-evolutionary processes [@Pantel2023].

Two existing statistical models are potentially well suited for analysis of eco-evolutionary data at the microevolutionary and metacommunity scale but also have some considerations before their application to complex community data. Lasky et al. (2020) [@Lasky2020] developed an integrated reaction norm model, linking genetic, phenotypic, and demographic processes, and Benito Garzón et al. (2019) [@Garzon2019] presented a species distribution model with local adaptation and phenotypic plasticity ($\Delta$ SDMs). The model of Lasky et al. is an excellent candidate for spatially complex population dynamics, and awaits an extension to the multispecies level. The $\Delta$ SDM models described by Benito Garzón et al. truly disentangle the role that plastic and evolutionary trait divergence can play in species distributions, but they also require common garden experimental data across sites to estimate these effects. Observational data collected in metacommunity surveys may only have observational data across sites or time points, as large-scale experimental measures of reaction norms required by the $\Delta$ SDM model may be difficult to obtain.

One particular metacommunity statistical model, the Hierarchical Modelling of Species Communities (HMSC) framework [@Ovaskainen2017HMSC] [@Tikhonov2020], explicitly considers the multitude of processes that can structure community composition across time and space [@Leibold2022] and is thus well suited to consider analysis of metacommunity eco-evolutionary data. This model of environmental filtering via variation and covariation in how species respond to their environment considers the impacts of (fixed) trait values and phylogenetic relationships, and uses latent variables to account for the numerous unobserved environmental and spatial features that are difficult to exhaustively measure in community surveys. The associated R package has immense flexibility, and has been applied to study environmental and abiotic drivers of diverse assemblages of organisms (e.g. [@Sandal2022] [@Weigel2023]). HMSC is potentially useful for studying eco-evolutionary drivers of community structure, but the model has not previously been applied to include heritable changes in trait values over time.

In this study, we use simulations of population dynamics, environmental drivers, species interactions, and trait evolution in populations, communities, and at the metacommunity scale to generate realistic time series of population sizes and mean trait values. We then use HMSC to analyze the resulting data, to determine whether HMSC can successfully estimate the impacts of trait evolution for community composition, and to generate relative contributions of fixed environmental, spatial, and trait evolution drivers as well as random effects of spatial and temporal covariance among species. In order to determine whether HMSC is useful for hypothesis testing about the drivers of eco-evolutionary dynamics in communities in a structured landscape, we XXX SIMULATIONS XXX to determine whether these potential drivers altered the importance of evolution for metacommunity composition. We hypothesized that XXX. We also XXX EMPIRICAL XXX.

Methods
================================================================================

## Linear statistical models for one- and two-species models of growth, competition, and trait evolution

We evaluated whether linear regression models can appropriately capture the dynamics that drive species abundances in more complex communities. We fit data generated from simulation models with increasing complexity to Bayesian linear regression models. These simulation models were (i) a single-species model of discrete-time logistic population growth, (ii) expanded to include a randomly fluctuating environmental property $E$ that impacts the species' population growth rate, (iii) a two-species model of logistic growth and competition that was then (iv) expanded to consider a randomly fluctuating environmental property, (v) and further expanded to consider evolution in a trait $x$ that is selected by the environmental property, and finally (vi) the model of two-species growth, competition, environmental change, and evolution in a spatially structured (10-patch) habitat.

For linear regression models to fit to the simulated population size data, we began with models using population size and environment as fixed effects ($N_t \sim N_{t+1} + E_t + E_t^2$), and progressed to a mixed model using environment and trait evolution as fixed effects ($N_t \sim E_t + E_t^2 + |\Delta x|$) and time and spatial distance as random effects. For consistency, all linear regression models were implemented using HMSC (Hierarchical Modelling of Species Communities [@Ovaskainen2017HMSC] [@Tikhonov2020]). HMSC uses a latent variable approach as an alternative to directly modelling species interaction coefficients, estimating a reduced number of linear combinations of species abundances that best predict future population size values. For linear models fit to data generated in simulations *iii*, *iv*, *v*, and *vi* above, the time random effect was implemented to capture species interactions (as part of the temporal species association matrix estimated by the model), and the spatial random effect was implemented in model *vi* to capture spatial covariance in abundances as a function of distance between sites. Full methodological details are given in Appendix S1.

## Evolution in metacommunities and HMSC

We simulated growth and competition dynamics for a multi-species assemblage in a patchy landscape, with site variation in one environmental property.

### Environmental variation

Population growth for species in the metacommunity simulation follows a Leslie-Gower model (a discrete-time version of a Lotka-Volterra model [@Beverton1957] [@Leslie1958]). We consider the impact of trait evolution for growth using a discrete time quantitative genetic model of evolutionary rescue [@Gomulkiewicz1995]. The model for population size is as follows:

$$ N_{i,t+1} =\frac{\hat{W}e^ \frac{-[(\frac{w+(1-h^2)P} {P+w})(E-x_{i,t})]^2}{2(P+w)}N_{i,t}} {1 + \alpha_{ii}N_{i,t} + \sum_{j \neq i}^{S} \alpha_{ij}N_{j,t}} $$
where $N_i,t$ is the population size of species *i* at time *t*, \( \hat{W} \) is calculated as \( \hat{W}=W_{max}\sqrt(\frac{w}{P+w}) \), \(W_{max}\) is the species' maximum per-capita growth rate, *w* is the width of the Gaussian fitness function (which determines the strength of selection, as increasing values indicate a weaker reduction in fitness with distance from optimum trait value), *P* is the width of the distribution of the phenotype *x*, $h^2$ is the heritability of the trait *x*, *E* is the local environmental optimum trait value, $x_{i,t}$ is the trait value of species *i* at time *t*, $\alpha_{ii}$ is the intraspecific competition coefficient (the per capita impact of species *i* on itself) and $\alpha_{ij}$ is the interspecific competition coefficient. Populations have a critical density $N_c$, below which the population is subject to extinction due to demographic stochasticity at a probability of *p* [@Gomulkiewicz1995].


Results
===============================================================================

## Linear statistical models for one- and two-species models of growth, competition, and trait evolution

Linear regression models can estimate the impacts of intra- and interspecific competition, environmental variation, and trait evolution for population size in simple one- and two-species models of population growth. In simple cases (one-species models, linear and quadratic relationships between environment and population size), the regression coefficients successfully reflect the direction and magnitude of the relationships included in the non-linear simulation model (the impact of $N_t$ and $E_t$ for $N_{t+1}$). However, as the processes included in the simulation models become slightly more complex (competition with a second species, spatial structure), regression coefficients were useful for prediction but not for recreating the underlying processes driving changes in population size (Appendix S1; REF?).

To consider species interactions and environmental fluctuations (model *iv*; Figure 1a \@ref(fig:fig1)), we used the latent variable modelling approach implemented in HMSC. The linear model successfully estimated coefficients for fixed environmental effects that captured the impacts of environment for population size in both species (Figure 1b,c) and partitioned the relative importance of the environment and species interactions for species abundances (Figure 1d). The model coefficients could also predict the population size values over time (Figure 1a), though the quality of this prediction varied depending on the level of environmental noise (Appendix S1).

```{r, out.width="90%", include=TRUE, fig.align="center", fig.cap=c("your caption"), echo=FALSE}
knitr::include_graphics("../output/fig1.pdf")
```


The HMSC model was successfully modelled changes in species abundances when trait evolution was introduced (model *v*; Figure 1e \@ref(fig:fig1)). Fixed effect coefficients for the environmental driver and the total trait change from one time to the next (|\Delta x|) matched well with observed data (Figure 1f,g). The model estimated the relative importance of the environmental driver, trait evolution in both species, and species interactions (Figure 1h), and predicted the observed species abundances well (Figure 1e). Spatial structure was also reproduced in the coefficients of the linear HMSC model by using spatial random effects to capture covariance in species abundances as a function of spatial distance between sites in the 10-patch simulation model (model *vi*; Figure 1i,j,k \@ref(fig:fig1)). The variation partition successfully reflects separation of environmental drivers from trait evolution, species interactions, and spatial random effects (Figure 1l), and we show in Appendix S1 that the relative importance of trait evolution increases when we use a simulation model with less spatial and environmental variation and weak species interactions.

## Evolution in metacommunities and HMSC

We simulated growth and competition dynamics for a multi-species assemblage in a patchy landscape, with site variation in one environmental property.

### Environmental variation

Population growth for spec

Discussion
================================================================================

XXXXX


References
===============================================================================

<div id="refs"></div>

<a name="appendix">Figures & Tables</a>
================================================================================

