---
title: "Detection of eco-evolutionary dynamics in metacommunities using Joint Species
  Distribution Models"
author:
- "Jelena H. Pantel^[Laboratoire Chrono-environnement,UMR 6249 CNRS-UFC, 16 Route
  de Gray, 25030 Besançon cedex, France, jelena.pantel@univ-fcomte.fr]"
- "Ruben J. Hermann^[University of Duisburg-Essen, Universitätsstraße 5, 45141 Essen,
  Germany, ruben.hermann@uni-due.de]"
date: "`r format(Sys.time(), '%d %B, %Y')`"
output:
  word_document:
    reference_docx: docx_template.docx
  bookdown::html_document2: default
  bookdown::pdf_document2:
    toc: no
    fig_caption: yes
    latex_engine: xelatex
    extra_dependencies: float
    number_sections: yes
  pdf_document:
    toc: no
header-includes:
- \usepackage{fancyhdr}
- \pagestyle{fancy}
- \fancyhead[L]{Author Pantel and Hermann}
- \fancyhead[R]{Detection of ecoevo dynamics}
- \usepackage{lineno}
- \linenumbers
linestretch: 2
link-citations: yes
linkcolor: blue
csl: "the-american-naturalist.csl"
bibliography: refs.bib
---

```{r setup,echo=FALSE,warning=FALSE,message=FALSE}
library(Hmsc)
library(knitr)
library(kableExtra)
library(formatR)
library(rmarkdown)
```

\newpage

# Title

Detection of eco-evolutionary dynamics in metacommunities using Joint Species Distribution Models

# Abstract

XXXXX

# Introduction

Analysis of biodiversity at large spatial scales can be complex, because numerous environmental properties, spatial structure and connectivity, interactions between species, as well as stochastic processes such as drift and priority effects operate and determine observed patterns. Statistical models in ecology have made substantial progress in reflecting these numerous ecological processes at multiple scales. There are now a diversity of joint species distribution models [JSDMs; @Pollock2014] [@Pichler2021] and multi-species occupancy models [@Devarajan2020] [@Poggiato2021], which can incorporate these diverse components that structure variance in biological data. Many of these models have implementations in R packages (e.g. spOccupancy, [@Doser2022]), and are used to measure the relative importance of different ecological drivers in e.g. communities of fungi, birds, mammals, phytoplankton, fish, and other taxa, from local to continental scales (e.g. [@Abrego2020] [@Antao2022] [@Keppeler2022]).

Despite the variety of components implemented in these models, there are some additional drivers of community composition which are not currently reflected. The dynamics of populations and communities, and resulting biodiversity patterns, are also influenced by the genotypic and phenotypic properties of organisms. Although traits can be considered in some JSDMs, they typically require fixed values for species' traits (e.g. [@Ovaskainen2017HMSC] [@Tikhonov2020]; but see @Abrego2024). However, ecological dynamics can impact selection on genes and phenotypes, potentially resulting in interactions or feedback loops between ecological and evolutionary processes [@Lion2018] [@Barbour2022]. Although an increasing number of theoretical and empirical studies have evaluated the impact of evolutionary dynamics for community assembly and metacommunity dynamics [@Loeuille2008] [@Osmond2013] [@Pantel2015] [@Toju2017], analytical methods to consider this additional complexity lag behind the scope of data collected in eco-evolutionary studies. Some methods exist that ultimately apply ANOVA to partition variance in traits or growth rates across contributions of ecological and evolutionary components ([@Hairston2005] [@Govaert2016]; methods for discrete and continuous analyses are described in @Ellner2011). These methods are more frequently cited for their potential rather than for their actual application to empirical data (but see [@terHorst2014] [@Hiltunen2014]), because this approach is ultimately limited by the same requirements that can limit application of ANOVA - data should follow a normal distribution, the processes that structure the target response variable must be linear and additive, and complicated structuring mechanisms must be reduced into one or a few target categories labeled as 'ecology' or 'evolution'. Flexible models, which can better reflect the mechanisms and structure of the diverse kinds of data that most studies of eco-evolutionary dynamics will collect, are needed for rigorous inference about eco-evolutionary processes [@Pantel2023].

Two existing statistical models are potentially well suited for analysis of eco-evolutionary data at the microevolutionary and metacommunity scale but also have some considerations before their application to complex community data. Lasky et al. (2020) [@Lasky2020] developed an integrated reaction norm model, linking genetic, phenotypic, and demographic processes, and Benito Garzón et al. (2019) [@Garzon2019] presented a species distribution model with local adaptation and phenotypic plasticity ($\Delta$ SDMs). The model of Lasky et al. is an excellent candidate for spatially complex population dynamics, and awaits an extension to the multispecies level. The $\Delta$ SDM models described by Benito Garzón et al. truly disentangle the role that plastic and evolutionary trait divergence can play in species distributions, but they also require common garden experimental data across sites to estimate these effects. Observational data collected in metacommunity surveys may only have survey data across sites or time points, as large-scale experimental measures of reaction norms required by the $\Delta$ SDM model may be difficult to obtain.

One particular metacommunity statistical model, the Hierarchical Modelling of Species Communities (HMSC) framework [@Ovaskainen2017HMSC] [@Tikhonov2020], explicitly considers a multitude of ecological processes that can structure community composition across time and space [@Leibold2022] and is potentially well suited to consider analysis of metacommunity eco-evolutionary processes. This model infers the role of environmental filtering via variation and covariation in how species respond to their environment, considers the impacts of (fixed) trait values and phylogenetic relationships, and uses latent variables to account for the numerous unobserved environmental and spatial features that are difficult to exhaustively measure in community surveys. Previous studies have demonstrated that this statistical model can reflect the diverse array of community assembly and metacommunity processes that structure species abundances (i.e. species sorting, environmental filtering, priority effects; [@Little2018] [@Ovaskainen2020] [@Leibold2022]). The associated R package has been applied to study environmental and abiotic drivers of diverse assemblages of organisms (e.g. [@Sandal2022] [@Weigel2023]). HMSC is potentially useful for studying eco-evolutionary drivers of community structure, but the model has not previously been applied to include heritable changes in trait values over time.

In this study, we use simulations of population dynamics, environmental drivers, species interactions, and trait evolution in populations, communities, at the local and metacommunity scale, to study eco-evolutionary dynamics. After generating simulated time series of population sizes and mean trait values, we then use HMSC to analyze the resulting data, to determine whether HMSC can successfully estimate the impacts of trait evolution for community composition, and to generate relative contributions of fixed environmental and trait evolution drivers as well as random effects of spatial and temporal covariance among species. We asked three main questions: (i) can linear statistical models (like HMSC) accurately capture the impact of non-linear population, community, and evolutionary dynamic processes?, (ii) can HMSC be used to evaluate hypothesized drivers of the relative importance of evolution for community composition at the metacommunity scale?, and (iii) can we better understand the role of evolution for community dynamics in experimental systems using HMSC?

# Methods

## Linear statistical models for one- and two-species models of growth, competition, and trait evolution

We evaluated whether linear regression models can appropriately capture the dynamics that drive species abundances in relatively simple populations and communities. We fit data generated from simulation models with increasing complexity (\@ref(tab:tab1)) to Bayesian linear regression models. These simulation models were (A) a single-species model of discrete-time logistic population growth, (B) expanded to include a randomly fluctuating environmental property $E$ that impacts the species' population growth rate, (C) a two-species model of logistic growth and competition that was then (D) expanded to consider a randomly fluctuating environmental property, (E) and further expanded to consider evolution in a trait $x$ that is selected by the environmental property, and finally (F) the model of two-species growth, competition, environmental change, and evolution in a spatially structured (10-patch) habitat.

```{r tab1, echo=FALSE}
df <- data.frame(model=c("1 species, logistic","1 species, logistic, env","2 species, logistic, competition","2 species, logistic, competition, env","2 species, logistic, competition, env, trait evolution","2 species, logistic, competition, env, trait evolution, spatial structure"), row.names=LETTERS[1:6])

df$formula <- c("$N_{i,t+1} = \\frac{r_i N_{i,t}}{1+\\alpha_{ii} N_{i,t}}$",
                "A, $r_i = \\hat{W}e^{-(E_t - x_{i,t})^2}$",
                "$N_{i,t+1} = \\frac{r_i N_{i,t}}{1+\\alpha_{ii} N_{i,t} + \\alpha_{ij} N_{j,t}}$",
                "C, $r_i = \\hat{W}e^{-(E_t - x_{i,t})^2}$",
                "C, $r_i = \\hat{W}e^\\frac{-[(\\frac{w+(1-h^2)P}{P+w})(E_t - x_{i,t})]^2}{2 (P+w)}$",
                "E, $N_{i,k,t+1} = N_{i,k,t} + \\sigma_k + \\sigma_{kl}$")
kable(df, "latex", align="l", booktabs=TRUE, escape = F, caption = 'Models for population growth')  %>%
footnote(general=c("$i,j=$","species,","$t=$","time,","$k,l=$","site"),
         number=c("See Appendix S1 for parameter values used in simulation"),
         footnote_as_chunk=TRUE, 
         escape=FALSE)
```

To fit linear regression models to the simulated population size data, we began with models using population size (model *A*) and environment (including the second-order term, to better reflect species optimum values; model *B*= as fixed effects ($N_t \sim N_{t+1} + E_t + E_t^2$), and progressed to a mixed model using environment and trait evolution as fixed effects ($N_t \sim E_t + E_t^2 + |\Delta x|$; for simulation model *E*) and time (models *C*, *D*, *E*, and *F*) and spatial distance (model *F*) as random effects. For consistency, all linear regression models were implemented as hierarchical Bayesian models using HMSC (Hierarchical Modelling of Species Communities [@Ovaskainen2017HMSC] [@Tikhonov2020]). HMSC uses a latent variable approach as an alternative to directly modelling species interaction coefficients, estimating a reduced number of linear combinations of species abundances that best predict future population size values. For linear models fit to data generated in simulations *C*, *D*, *E*, and *F* above, the time random effect was implemented to capture species interactions (as part of the temporal species association matrix estimated by the model), and the spatial random effect was implemented in model *F* to capture spatial covariance in abundances as a function of distance between sites. Full methodological details are given in Appendix S1. All simulations were run in R (@R2024) and HMSC models were implemented using the `Hmsc` package (@Hmsc2022).

## Evolution in metacommunities and HMSC

### Simulation model

We also simulated growth and competition dynamics for a multi-species assemblage in a patchy landscape, with site variation in one environmental property, and with evolution in a trait that is selected on by the environment. Population growth for species in the metacommunity simulation followed a Leslie-Gower model (a discrete-time version of a Lotka-Volterra model [@Beverton1957] [@Leslie1958]). We considered the impact of trait evolution for growth using a discrete time quantitative genetic model of evolutionary rescue [@Gomulkiewicz1995]. The model for population size was:

$$ N_{i,t+1} =\frac{\hat{W}e^ \frac{-[(\frac{w+(1-h^2)P} {P+w})(E-x_{i,t})]^2}{2(P+w)}N_{i,t}} {1 + \alpha_{ii}N_{i,t} + \sum_{j \neq i}^{S} \alpha_{ij}N_{j,t}} $$ where $N_i,t$ is the population size of species *i* at time *t*, $\hat{W}$ is calculated as $\hat{W}=W_{max}\sqrt(\frac{w}{P+w})$, $W_{max}$ is the species' maximum per-capita growth rate, *w* is the width of the Gaussian fitness function (which determines the strength of selection), *P* is the width of the distribution of the phenotype *x*, $h^2$ is the heritability of the trait *x*, *E* is the local environmental optimum trait value ($x_{opt}$), $x_{i,t}$ is the trait value of species *i* at time *t*, $\alpha_{ii}$ is the intraspecific competition coefficient (the per capita impact of species *i* on itself) and $\alpha_{ij}$ is the interspecific competition coefficient. Populations have a critical density $N_c$, below which the population is subject to extinction due to demographic stochasticity at a probability of *p* [@Gomulkiewicz1995].

This model was used in Pantel & Becks [@Pantel2023] to evaluate the consequences of adaptive evolution for coexistence in a three-species system. To expand this model to a metacommunity, we consider the evolution of multiple species (*S* = 15) in a landscape of patches (*k* = 50). The patches have values of an environmental property $E \sim U(0,1)$ that determines the local optimum phenotype *E* = $x_{opt}$, i.e. where species experience their maximum absolute fitness $W_{max}$, and patches also have spatial locations *X* and *Y* (both drawn from $U(0, 1)$). The patches thus have a connectivity matrix **D** (here given by their Euclidean distance), as well as a connectivity matrix **C** that is a Gaussian function of **D** and species dispersal rate $d_i$.

We used the evolving metacommunity model to test whether landscape connectivity (the overall dispersal rate $d$) and the speed of evolution (varying $h^2$) influenced the relative importance of trait evolution ($|\Delta x|$) for metacommunity composition $N_{i,t,k}$ (evaluated as $V_{|\Delta x|}$, the proportion of variation in species abundances explained by $|\Delta x|$ as estimated by the HMSC model fit to the simulated data across $d$ and $h^2$ levels). We hypothesized that the overall importance of trait evolution ($V_{|\Delta x|}$) would decrease with increasing dispersal rate - at low dispersal rates, locally mal-adapted species are less likely to emigrate to more optimal patches and will experience strong local selection, leading to a large amount of trait evolution. Population sizes for initial, mal-adapted populations will also likely be lower, leading to reduced relative importance of density-dependent intra- and interspecific competition. We also hypothesized that heritability ($h^2$) would mediate the relative importance of dispersal rate - at higher $h^2$, species will evolve quickly and we expected the importance of trait evolution $V_{|\Delta x|}$ to depend on the amount of time that mal-adapted populations persist. Therefore, simulations with higher $h^2$ would have decreased importance of evolution $V_{|\Delta x|}$ for community composition. For the same reason, we hypothesized generally that, for a given dispersal rate $d$, the importance of evolution $V_{|\Delta x|}$ would decrease with increasing $h^2$.

All simulations used the same parameter values ($W_{max}$ = 2, *P* = 1, *w* = 2, $N_c$ = 100, *p* = 0.001), species interaction matrix ($\alpha_{ii} = 0.00125$; $\alpha_{ij} \sim U(0.0015,1)$), and initial conditions (initial richness $s_{0k} \sim Pois(0.75)$; initial population size $N_{0i} \sim Pois(10)$; initial degree of maladaptation $\beta_{i0} \sim Gamma(0.75,1)$, initial distance to the local patch's optimum phenotype $d_{i0}=\sqrt{B_{i0}(w+P)}$, initial trait value $x_{i0} = E_k - d_{i0}$; Appendix S4). Simulations were run across a range of dispersal values (identical for all species; *d* = 0, $10^{-9},10^{-8},10^{-7},10^{-6},10^{-5},10^{-4},10^{-3},10^{-2}$, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1) and heritability values (identical for all species: $h^2$ = 0, 0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1) for 200 time steps, producing records of $N_{ikt}$ and $x_{ikt}$.

### Statistical model

To determine the relative influence of environmental and spatial properties, intra- and interspecific population dynamics, and evolutionary dynamics for community structure, we applied hierarchical modelling of species communities (HMSC, [@Ovaskainen2017HMSC] [@Tikhonov2020]) to model the log-abundance at each site and year for each species. The log-abundance values were modeled as $ln(N_{ikt}) \sim N(L_{ikt},\sigma_2)$, where the linear predictors **L** are the sum of fixed and random effects **L^F^** + **L^R^** [@Tikhonov2020]. The fixed effects were $E_k$, $E_k^2$, and the absolute value of the change in each species' trait value from the previous time step to the next $|\Delta x_{ik,t-1 \to t}|$. Random effects included time (the day of the sample) and space. The spatial latent variables represent a spatial model, where species respond to some latent spatial predictor associated with the *xy* coordinates of the sites [@Tikhonov2020]. When there was no dispersal in the system ($d=0$), spatial random effects were not included in the model.

### Applying HMSC on empirical data

To test the applicability of HMSC on ecological and evolutionary effects in empirical data, we applied HMSC on the experimental data of Jewel & Bell (2023) [@Jewell2023]. The authors conducted a community selection experiments in which they grew four floating and short-lived aquatic plants (the angiosperms *Lemna minor* here designated as Lm, *Spirodela polyrhiza* designated as Sp and *Wolffia columbiana* designated as Wc, and the liverwort *Ricciocarpos natans* designated as Rn) under different environmental conditions for twelve weeks. The environmental conditions were defined by three different shade cover levels for the experimental pools, 3%, 12% and 50%, and different nutrient concentrations, with low (dissolved nitrogen (DN) = 200 $\mu g L^{-1}$, dissolved phosphorous (DP) = 10 $\mu g L^{-1}$, medium (DN = 800 $\mu g L^{-1}$, DP = 40 $\mu g L^{-1}$), and high (DN = 3200 $\mu g L^{-1}$, DP = 160 $\mu g L^{-1}$) concentrations of nutrients. They combined each shade and nutrient level, resulting in nine different environmental conditions. For the eleven weeks of the experiment, the frond size of ten individuals from each species, as well as population density were measured. To use the data with our HMSC analysis, we calculated the average trait change as $\Delta x = x_{ijt} - x_{ijt-1}$, where $x_{ij}$ described the mean trait of species $i$ in environment $j$ and week $t$.

To determine the relative importance of evolution for community composition, we fitted two HMSC models to the data: An 'evolution model' in which $\Delta \hat{x}$ of each species and environmental conditions were used as explanatory variables, and a 'no-evolution model', in which only the environmental conditions were included. Species density was log-transformed species before applying it to the model, therefore we could use normal distribution in the model. We also included replicates (*n* = 2) and time as random effects. For each model we ran four MCMC chains to generate 1000 samples of the posterior distribution for all model parameters. To evaluate the model-estimated role of evolution for predicted population size, we used the posterior distributions of the estimated effect size of $|\Delta x|$ multiplied by the observed $\Delta \hat{x}$ of each species,added to the model-estimated intercept.

# Results

## Linear statistical models for one- and two-species models of growth, competition, and trait evolution

Linear regression models can estimate the impacts of intra- and interspecific competition, environmental variation, and trait evolution for population size in simple one- and two-species models of population growth. In simple cases (one-species models, linear and quadratic relationships between environment and population size; models *A* and *B*), the regression coefficients can successfully reflect the direction and magnitude of the relationships included in the non-linear simulation model (the impact of $N_t$ and $E_t$ for $N_{t+1}$). However, as the processes included in the simulation models become slightly more complex (competition with a second species, spatial structure; models *C*, *D*, *E*, *F*), regression coefficients were useful for prediction but not for recreating the underlying processes driving changes in population size (Appendix S1).

To consider species interactions and environmental fluctuations (model *D*; Figure 1a \@ref(fig:fig1)), we used the latent variable modelling approach implemented in HMSC. The linear model successfully estimated coefficients for fixed environmental effects that captured the impacts of environment for population size in both species (Figure 1b) and partitioned the relative importance of the environment and species interactions for species abundances (Figure 1c). The model coefficients could also predict the population size values over time (Figure 1a), though the quality of this prediction varied depending on the level of environmental noise (Appendix S1).

```{r fig1, out.width="90%", include=TRUE, fig.align="center", fig.cap=c("Simulated and model-estimated values for population dynamics in two-species systems. Panels a, d, and g show log-population size (y-axis) over time (x-axis) for species $i$ and $j$ when they grow and compete for resources in sites with random environmental fluctuations over time. Species either have fixed (panel a) or evolving trait values (panels d, g) and are present in either 1 (panels a, d; simulation models D and E) or multiple sites (panel g; model F). Graphs in panels b, e, and h show predicted log-population size ($ln(N)$) for species 1 and 2 across gradients of the environment ($E + E^2$) and the change in trait values of species 1 ($|\\Delta x_1|$). Graphs in panels c, f, and i show the results of a variation partition after fitting data to the HMSC statistical model. The variation partition plot shows the proportion of variation explained by environmental, trait evolution ($|\\Delta x_1|$ and $|\\Delta x_2|$), time random, and spatial random predictors for $ln(N)$ for species 1 and 2. Error bars and shading represent the 95% highest probability density interval for posterior predictions."), echo=FALSE}
knitr::include_graphics("../output/fig1.pdf")
```

The HMSC model successfully modelled changes in species abundances when trait evolution was introduced (model *E*; Figure 1d \@ref(fig:fig1)). Trait evolution played a strong role for population size, especially the effect of $|\Delta x_1|$ for $ln(N_1)$ and $ln(N_2)$ (gradient plots in Figure 1e). The model estimated the relative importance of the environmental driver, trait evolution in both species, and species interactions (Figure 1f), and predicted the observed species abundances well (Figure 1d). Spatial structure was also reproduced in the coefficients of the linear HMSC model by using spatial random effects to capture covariance in species abundances as a function of spatial distance between sites in the 10-patch simulation model (model *F*; Figure 1g,h,i \@ref(fig:fig1)). The variation partition successfully reflected separation of environmental drivers from trait evolution, species interactions, and spatial random effects (Figure 1i), and we show in Appendix S1 that the relative importance of trait evolution increases when we use a simulation model with less spatial and environmental variation and weak species interactions.

## Evolution in metacommunities and HMSC

The simulation model followed population size and trait value dynamics for species in the landscape across 200 time steps (Figure 2a \@ref(fig:fig2)). Resulting biodiversity dynamics depended on the dispersal level and heritability values (Figure 2b \@ref(fig:fig2)). For dispersal, results mirrored some expectations from existing metacommunity models, i.e. that regional $\gamma$ diversity decreases with increasing dispersal level ([@Mouquet2003]; Figure 2b \@ref(fig:fig2)). The inclusion of trait heritability had a strong effect on community diversity, reflecting that local adaptation can rescue species and contribute to richness at the local and metacommunity scale. Regional species diversity ($\gamma$) increased with increasing $h^2$ levels (Figure 2b \@ref(fig:fig2)).

Analysis using the HMSC model for community time series across $d$ and $h^2$ levels indicated that the importance of evolution for community composition ($V_{|\Delta x|}$) decreased with increasing dispersal and with increasing heritability (Figure 2c,d,e \@ref(fig:fig2)). The importance of spatial random effects increased with dispersal rate, and overall the importance of evolution decreased with increasing $h^2$.

```{r fig2, out.width="90%", include=TRUE, fig.align="center", fig.cap=c("Importance of trait evolution for change in community composition in metacommunities. (a) We developed a simulation model for population growth, competition, and trait evolution in 15 species that inhabit a patchy landscape. The landscape has 50 sites (colored points in the XY space), with distinct local environments that select for diverse patterns of species diversity (size of points) and average trait values (color of points; results are shown at time $t=0, 10, 200$ for a simulation with $d=10^-3$ and $h^2 = 0.1$). (b) Regional metacommunity $\\Gamma$ diversity varied across levels of dispersal and heritability ($h^2$). (c) Variation partition plots after fitting metacommunity time series of species abundances $ln(N_{ikt})$ to HMSC models with environment, trait evolution, and random effects (space and time) as predictors. The top panel is for $h^2 = 0.1$, the middle panel is for $h^2 = 0.3$, and the bottom panel is for $h^2 = 0.9$."), echo=FALSE}
knitr::include_graphics("../output/fig2.pdf")
```

### Empirical data

The empirical data [@Jewell2023] showed the community dynamics of four plant species under different environmental conditions, additionally the front size of each species was measured to determine its trait change over time. We calculated the mean variance partitioning (VP) over all species, to evaluate the impact of evolutionary (change of the front size trait) and ecological (light and nutrient treatments) effects on changes on species density. Almost one fifth (18.6%) of the variance in species density could be explained by the variance in $\Delta \hat{x}$ (\@ref(fig:fig5)a). This is due to the reduction of the VP of the ecological effects by a total of 14.4% (from 37% to 26.6% for the amount of shading and from 31% to 27.1% for the nutrient concentrations) and the reduction of 4.2% of the random effects (from 18.7% to 15% for time and from 13.3% to 12.8% for replicates) (\@ref(fig:fig5)). While $\Delta \hat{x}$ of the species Lm, Sp and Wc showed a similarly low impact, on average 3.6%, $\Delta \hat{x}$ of species Rn showed a high impact on variance of species density, 7.8%, which can be explained by its high $\Delta \hat{x}$ changes over time (see Appendix S05). The high density intervals (HDI) of 95% estimated beta effects showed a negative correlation of Light shading with species density, while nutrient concentration correlated positively with density (\@ref(fig:fig5)c). Changes in $\Delta \hat{x}$ impacted densities differently. At one hand changes of species Rn frond size correlated negatively with is density, on the other hand changes in the frond size of species Wc showed a positive correlation with the other species Sp and Lm (\@ref(fig:fig5)c). The impact of evolution (of all species summed up), impacted species density over time, yet did not show a clear direction over the course of the experiment (\@ref(fig:fig5)d).

```{r fig3, out.width="90%", include=TRUE, fig.align="center", fig.cap=c("The estimated effects of species trait changes and environmental treatments. The variance partitioning of all fixed effects of the HMSC model (a) including trait changes with envrionmental treatments and (b) only including the environmental treamtents. The (b) high density interval of 95% showed a clear pattern for the different environmental treatments, yet differing results for trait changes and their impact on species density. This can be seen in (d) the shifting predictions of density over time due to the total trait changes of all species."), echo=FALSE}
knitr::include_graphics("../output/fig5.pdf")
```

# Discussion

*Main findings*
The analysis of community composition often requires statistical models that can consider the combined influences of environmental drivers and spatial connectivity. A more difficult, yet critical, element to incorporate is species interactions. Although the causal role of species interactions is difficult to interpret from statistical models (but see [@Dubart2022] [@Luo2022]) in systems with few species), joint species distribution models (JSDMs) have implemented spatial and temporal random effects that can effectively capture these species interactions using a latent variable approach (but note the resulting species covariance matrices will also reflect other processes besides species interactions; [@Poggiato2021]). Finally, one element that has not been fully implemented into joint species distribution models is dynamic trait evolution (see [@Garzon2019] for a model that does consider reaction norms from in-situ reciprocal transplant experiments). We used an existing JSDM and developed simulation models of evolving populations in increasing complex scenarios to determine whether the signal of trait evolution could be estimated from data and compared to other drivers of community structure. We also applied this statistical model (HMSC, hierarchical models of species distributions; [@Ovaskainen2017HMSC] [@Tikhonov2020]) to an existing empirical dataset (from the experimental study of community eco-evolutionary dynamics in aquatic plants of [@Jewell2023]) to show how HMSC can be used to model the changes in species composition over time as a function of environmental drivers, trait change, and temporal random effects. Overall, there is no barrier to including dynamic trait change as a fixed effect in a JSDM model, and the resulting effect size coefficient estimates can be used to (i) estimate the relative importance of trait evolution for community composition and (ii) predict the impacts of future trait change in the system. 

*Discussion regarding simple case studies*
At the same time, fixed effects for the impacts of trait evolution on population size should be interpreted with a great deal of caution. For example, our simulations reveal instances where trait evolution may occur at the same time as large changes in population size, and this co-occurrence isn't always causal (e.g. when selection is weak but trait heritability is high, and species are introduced at low initial population size). In this instance, the regression coefficient from the model may be negative and strong. By using simulations of trait evolution and population growth in simple systems (1- and 2-species systems, at single and simplified multi-patch scales; \@ref(fig:fig1); Appendix S1), we can better interpret effect size estimates. However, these precautions are not unique to including trait evolution as a predictor of community composition. For example, previous research has shown that the effect size coefficients from linear and AR models using $N_{i,t-1}$ to predict $N_{i,t}$ or $N_{j,t}$ are not equivalent to the intra- and interspecific interaction coefficients that determine species density dependence and should not automatically be used to interpret the direction and magnitude of species interactions ([@Kloppers2013] [@Certain2018] [@Muhlbauer2020] [@Olivenca2021]). Our findings for using HMSC to estimate the impact of trait evolution ($|\Delta x_i|$) for species abundances are similar - (i) the relative importance of trait evolution estimated by a variation partition accurately reflects the relative strength of simulated species interactions vs. magnitude of trait change and (ii) the coefficients greatly increase the predictive ability of the statistical model for simulated data (Appendix S1). HMSC is therefore a useful tool for understanding the role that evolution can play in community dynamics.

*Discussion regarding metacom simulations*
This was also observed when using HMSC to evaluate the importance of evolution for metacommunity structure, using simulated data from a model of population growth, competition, dispersal, and trait evolution in a spatially structured habitat (Figure 2a \@ref(fig:fig1); Appendix S4). First of all, the simulations themselves revealed an intriguing role of adaptive evolution for rescuing individual species from extinction (i.e. [@Gomulkiewicz1995]) and as a consequence greatly increasing biodiversity at the local and the landscape scale (i.e. community rescue, [@Low2015] [@Bell2019]; Figure 2b \@ref(fig:fig2)). The recovery of species via local adaptation had consequent effects for coexistence, as species that were initially mal-adapted could recover in population size and exert strong competitive effects on other species. Because HMSC can accurately estimate the relative importance of trait evolution for community composition (Appendix S4), we were thus able to test hypotheses about the drivers of increased or decreased impacts of trait evolution for community structure. Our simulation models fit to HMSC confirmed our hypotheses that the importance of evolution would decrease with increasing landscape connectivity (modeled as species dispersal rate $d$) and the overall speed of adaptive evolution (modeled as heritability $h^2$). Previous theoretical models have been shown that these two properties can have a strong influence on resulting community composition ([@Urban2008] [@Loeuille2008] [@Urban2009] [@DeMeester2016]), but our demonstration that the HMSC statistical model can accurately estimate $R^2$ values for the relative importance of trait evolution can allow hypotheses from theoretical models to be tested using empirical data and compared across different study systems.

*Discussion regarding empirical data*
By implementing the trait changes of the four different species (Lm, Rn, Sp and Wc) in the HMSC model we were able to infer the relative impact of the trait changes on the community dynamics. This application shows is advantages over previous methods,which estimate the impact of trait changes (thus evolution), as it is not depended on the data structure and simplicity [@Hairston2005] [@Ellner2011], can be implemented on multiple species[@Lasky2020] and does not require additional common garden experiments [@Garzon2019]. Therefore this allows for a straight forward implementation of HMSC on a variety of data structures as lone as a clear response variable (e.g. species density) and ecological (e.g. environmental conditions) as well as evolutionary (e.g. heritable trait change) are measured. An additional advantage of HMSC, and JSDMs in general, is the ability to use the empirical data to predict a species response to changes in the environmental variables, even beyond the measured values [@Tikhonov2020] [@Ovaskainen2015]. With the implementation this approach can be combined with the fixed effects to answer further questions, such as: How does trait change impact the species density under changing environmental conditions? And how does the impact of trait changes depend on specific environmental conditions? 

At the same time a correct interpretation of such predictions as well es the estimated posterior beta distribution require a thorough understand of the experimental system. For example we often observed a negative correlation between trait change and species density in our simulations and also in species Rn of the empirical data, yet they derived from different mechanisms. In our simulations trait changes were greater when a species trait was further away from the optimal trait value, this greater distance resulted also in lower fitness and thus lower population size, while a species with higher fitness and thus higher population size had smaller trait changes. Therefore, HMSC correlates a high trait changes with a low population size and thus estimates a negative beta effect. Such effects can be expected in empirical data of i.e. evolutionary rescue experiments, where a species has to adapt to a deleterious environment before going extinct [@Gomulkiewicz1995][@Bell2009]. In the empirical data this negative correlation was the result of the species Rn investing into a larger frond size (Fig. SX), this likely lead to a lower investment into reproduction, while still maintaining a similar total frond area on the water surface to capture the reduced light.

*Takehome points and conclusions*
There has been a strong general interest in estimating effect sizes of evolution for ecological processes [@Hairston2005]; [@Ellner2011]; [@Pantel2015]; [@Barbour2022]), but in many empirical studies these estimates focus on one or a few species where the impacts of their evolution is strong enough to play a clear and detectable role in the corresponding ecological response variable of interest (e.g. population growth rate, community diversity, likelihood of food web collapse). There is no general knowledge for overall distributions of effect sizes for the impact of evolution for species composition in communities, nor are there studies that evaluate drivers of shifts in these distributions of effect sizes in complex communities (see [@Hermann2024] for estimates in a predator-prey system). We present here distributions of effect sizes of evolution for species abundances (i.e. Figure 1c,f,i \@ref(fig:fig1); Figure 2c \@ref(fig:fig2); Figure 3c \@ref(fig:fig3)) in simulated and experimental aquatic plant communities, and expect that these effect size coefficients can be useful for better understanding the relative importance of evolution as a driver of community composition.

An increasing number of studies collect and analyze high-resolution time series of populations and trait values for single [@Rudman2022] and multiple [@Thomas2018] [@Gibert2023] [@Han2023] [@Shen2023] species. With increasing access to ecological and evolutionary data comes an increase in the number of interacting processes that should be considered to identify causal mechanisms that structure this data [@Pantel2023]. HMSC, and joint species distribution models generally, can be expanded to further include dynamic trait change that occurs during the course of community ecological studies. Two critically important additions that will increase the application to field or experimental observation data will be (1) bidirectional causality and eco-evolutionary feedbacks and (2) the relative contribution of plasticity and genotypic evolution for trait change (i.e. [@Gibert2022]). It will also be necessary to improve knowledge of the distributions of eco-evolutionary effects and how these are influenced by causal processes that change the speed of evolution or the strength of species interactions. Although these interacting eco-evolutionary dynamic processes represent an increased degree of complexity for data analysis and interpretation, these mechanistic processes can be translated into simpler statistical models to estimate ecological and evolutionary drivers of community composition.

# References

::: {#refs}
:::

# <a name="appendix">Figures & Tables</a>
 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7 7